gooddata-sdk/gooddata_sdk/catalog/declarative/identifier.py

```diff
@@ -0,0 +1,46 @@
+# (C) 2022 GoodData Corporation
+from __future__ import annotations
+
+from typing import Any, Type, TypeVar
+
+from gooddata_metadata_client.model.grain_identifier import GrainIdentifier
+from gooddata_metadata_client.model.reference_identifier import ReferenceIdentifier
+from gooddata_metadata_client.model.workspace_identifier import WorkspaceIdentifier
+from gooddata_sdk.catalog.entity import CatalogTypeEntity
+
+T = TypeVar("T", bound="CatalogIdentifierBase")
+
+
+class CatalogIdentifierBase:
+    def __init__(self, id: str):
+        self.id = id
+
+    @classmethod
+    def from_api(cls: Type[T], entity: dict[str, Any]) -> T:
+        return cls(entity["id"])
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, CatalogIdentifierBase):
```

Comment:
`__eq__` is not overriden later, so it allows to compare descendants and get True on unrelated entities.